AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_component_validation.py

```diff
@@ -215,3 +217,41 @@ def validate_asset_property(property_name: str, get_asset_value: typing.Callable
     assert expected_asset.id.is_equal(expected_asset.id, set_value), \
         f"Error: The {component_name}'s {property_name} property failed to properly set the mesh. Asset Id: " \
         f"{expected_asset.id} was expected but Asset Id: {set_value} was retrieved from \"{expected_asset.get_path()}\""
+
+
+def validate_script_canvas_graph_file(get_script_canvas_components: typing.Callable, source_handle: str, old_value: str,
```

Comment:
Validators should be both a setter & a getter action and follow the patterns. The description even implies this, but is only validating that something has been set. This will also prevent needing to pass in the old value and setting up for testing the behavior outside of the validator. The validator for asset properties are probably the best to compare this particular validator to.

Validator expectations:

- Takes a getter and setter method in (these will hold the reference to the correct component since determining which component to validate should be done outside the validator)
- Takes the asset to set (Path in this case? Or you can pass a source handle if you prefer that method)
- Guard conditions executed
- Builds the asset correctly if using the path
- Captures the currently set asset
- Sets the Asset
- Validates the asset has been set.

Reply:
I think the validation functions now statisfy these expectations. both validation functions now take a set/get function as their parameters. They capture the old value we are testing by using the get function then compare it against the new value that is changed with the set function.